src/main/java/com/eternalcode/parcellockers/gui/ParcelListGUI.java

```diff
@@ -0,0 +1,81 @@
+package com.eternalcode.parcellockers.gui;
+
+import com.eternalcode.parcellockers.configuration.implementation.PluginConfiguration;
+import com.eternalcode.parcellockers.parcel.Parcel;
+import com.eternalcode.parcellockers.parcel.ParcelMeta;
+import com.eternalcode.parcellockers.parcel.ParcelRepository;
+import dev.triumphteam.gui.builder.item.ItemBuilder;
+import dev.triumphteam.gui.guis.Gui;
+import dev.triumphteam.gui.guis.GuiItem;
+import dev.triumphteam.gui.guis.PaginatedGui;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.minimessage.MiniMessage;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.Server;
+import org.bukkit.entity.HumanEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemFlag;
+
+public class ParcelListGUI {
+
+    private final int[] cornerSlots = {0, 8, 45, 53};
+    private final int[] borderSlots = {1, 2, 3, 4, 5, 6, 7, 9, 17, 18, 26, 27, 35, 36, 44, 46, 47, 48, 49, 50, 51, 52};
+    private final Server server;
+    private final MiniMessage miniMessage;
+    private final PluginConfiguration config;
+    private final ParcelRepository repository;
+
+    public ParcelListGUI(Server server, MiniMessage miniMessage, PluginConfiguration config, ParcelRepository repository) {
+        this.server = server;
+        this.miniMessage = miniMessage;
+        this.config = config;
+        this.repository = repository;
+    }
+
+    public void showParcelListGUI(Player player) {
+        GuiItem backgroundItem = this.config.guiSettings.mainGuiBackgroundItem.toGuiItem(this.miniMessage);
+        GuiItem cornerItem = this.config.guiSettings.cornerItem.toGuiItem(this.miniMessage);
+        GuiItem closeItem = this.config.guiSettings.closeItem.toGuiItem(this.miniMessage);
+
+        PaginatedGui gui = Gui.paginated()
+                .title(this.miniMessage.deserialize(this.config.guiSettings.parcelListGuiTitle))
+                .disableAllInteractions()
+                .rows(6)
+                .create();
+
+
+        for (int slot : this.cornerSlots) {
+            gui.setItem(slot, cornerItem);
+        }
+        for (int slot : this.borderSlots) {
+            gui.setItem(slot, backgroundItem);
+        }
+
+        for (Parcel parcel : this.repository.findAll()
```

Comment:
w tym przypadku przydała by się metoda w ParcelRepository#findParcelForUser

Reply:
Fakt, uwzględnie to przy reimplementacji bazy danych oraz dodam taska na Jirę